client/dashboard/sites/staging-site-delete-modal/test/index.test.tsx

```diff
@@ -0,0 +1,258 @@
+		createSuccessNotice: mockCreateSuccessNotice,
+	} ),
+	combineReducers: jest.fn( ( reducers ) => reducers ),
+	createReduxStore: jest.fn(),
+	register: jest.fn(),
+	createSelector: jest.fn(),
+} ) );
+
+jest.mock( '@tanstack/react-router', () => ( {
+	...jest.requireActual( '@tanstack/react-router' ),
+	useNavigate: () => mockNavigate,
+} ) );
+
+jest.mock( '@automattic/api-queries', () => ( {
+	siteByIdQuery: jest.fn( () => ( {
+		queryKey: [ 'site-by-id' ],
+		queryFn: () => Promise.resolve( {} ),
+	} ) ),
+	stagingSiteDeleteMutation: jest.fn( () => ( {
+		mutationFn: () => Promise.resolve( {} ),
+	} ) ),
+} ) );
+
+jest.mock( '@tanstack/react-query', () => ( {
+	QueryClient: jest.fn().mockImplementation( () => ( {} ) ),
+	QueryClientProvider: ( { children }: { children: React.ReactNode } ) => children,
+	useQuery: jest.fn( () => ( { data: { ID: 1, slug: 'production-site' }, isLoading: false } ) ),
+	useMutation: jest.fn( () => ( {
+		mutate: mockMutate,
+		isPending: false,
+		error: null,
+	} ) ),
+} ) );
+
+const createMockSite = ( options = {} ): Site =>
+	( {
+		ID: 2,
+		slug: 'staging-site',
+		name: 'Staging Site',
+		URL: 'https://staging-site.wordpress.com',
+		is_wpcom_staging_site: true,
+		capabilities: {
+			manage_options: true,
+		},
+		options: {
+			wpcom_production_blog_id: 1,
+			...options,
+		},
+	} ) as Site;
+
+const mockStagingSite = createMockSite();
+const mockStagingSiteWithoutProductionId = createMockSite( {
+	wpcom_production_blog_id: undefined,
+} );
+
+// Helper functions
+const getButton = ( name: string ) => screen.getByRole( 'button', { name } );
+const renderModal = ( site: Site, onClose = jest.fn() ) =>
+	render( <StagingSiteDeleteModal site={ site } onClose={ onClose } /> );
+
+describe( 'StagingSiteDeleteModal', () => {
+	beforeEach( () => {
+		jest.clearAllMocks();
+	} );
+
+	describe( 'Modal Display', () => {
+		test( 'renders modal with correct title and content', () => {
+			renderModal( mockStagingSite );
+
+			expect( screen.getByRole( 'dialog', { name: 'Delete staging site' } ) ).toBeInTheDocument();
+			expect(
+				screen.getByText(
+					'Are you sure you want to delete this staging site? This action cannot be undone and will permanently remove all staging site content.'
+				)
+			).toBeInTheDocument();
+		} );
+
+		test( 'displays cancel and delete buttons', () => {
+			renderModal( mockStagingSite );
+
+			expect( getButton( 'Cancel' ) ).toBeInTheDocument();
+			expect( getButton( 'Delete staging site' ) ).toBeInTheDocument();
+		} );
+
+		test( 'returns null when no production site ID is provided', () => {
+			const { container } = renderModal( mockStagingSiteWithoutProductionId );
+
+			expect( container.firstChild ).toBeNull();
+		} );
+	} );
+
+	describe( 'User Interactions', () => {
+		test( 'calls onClose when cancel button is clicked', () => {
+			const mockOnClose = jest.fn();
+			renderModal( mockStagingSite, mockOnClose );
+
+			fireEvent.click( getButton( 'Cancel' ) );
```

Comment:
When looking at other existing tests in `client/dashboard`, I have noticed that we rely more on `user.click` than `fireEvent.click`:

```suggestion
			await user.click( getButton( 'Cancel' ) );
```
But maybe there's a specific reason we are using `fireEvent.click` here?

Reply:
I made changes in https://github.com/Automattic/wp-calypso/pull/105770/commits/bdf0bae04f5bd402f6efec5ce724a8116084d8fe for consistency but it should not make any difference for these tests.

Reply:
Thanks! 🙂